packages/@sanity/base/src/change-indicators/overlay/ConnectorsOverlay.tsx

```diff
@@ -176,122 +90,60 @@ export const ConnectorsOverlay = React.memo(function ConnectorsOverlay(props: Pr
           return (
             <React.Fragment key={`field-${field.id}`}>
               {change && (
-                <g onClick={onConnectorClick} className={connectorClassName}>
-                  <Connector
-                    from={connectorFrom}
-                    to={connectorTo}
+                <>
+                  <g
+                    onClick={onConnectorClick}
                     onMouseEnter={() => setHovered(field.id)}
                     onMouseLeave={() => setHovered(null)}
-                    clampLeft={fieldClampConnector}
-                    clampRight={changeClampConnector}
-                    verticalCenter={verticalLineLeft! + 3}
-                  />
-
-                  {/* arrow left top */}
-                  {drawArrowLeftTop && (
-                    <Arrow
-                      top={fieldClampConnector.top}
-                      left={connectorFrom.left}
-                      length={5}
-                      wingLength={8}
-                      direction="n"
+                  >
+                    <Connector
+                      from={{rect: field.rect, bounds: field.bounds}}
+                      to={{rect: change.rect, bounds: change.bounds}}
+                      focused={hasFocus}
+                      hovered={hasHover || isHoverConnector}
+                      revertHovered={hasRevertHover}
                     />
+                  </g>
+                  {DEBUG_LAYER_BOUNDS && (
+                    <g style={{pointerEvents: 'none'}}>
+                      <rect
```

Comment:
Perhaps this could be made into a sub-component as well to make the main component a bit more expressive? :) 

```typescript
const DebugRects = (field, change) => (
  <g style={{pointerEvents: 'none'}}>
    <Rect points={field.bounds} />
    <Rect points={field.rect} />
    ...
  </g>
)

const SvgRect = (points: Rect) => (
  <rect x={points.left} .... />
)
```

Reply:
That's a very good idea. Let me see if I can sneak in a quick refactor.